### PR TITLE
[TR] PIM-5006 : No way to generate API Key on EE

### PR DESCRIPTION
**Pull request by jmleroux**
We need to use the User repository service in order to retrieve an instance of Pim User or PimEnterprise User.

### CODE REVIEW

File: src/Oro/Bundle/UserBundle/Controller/UserController.php [hunk 1]
```diff
@@ -50,9 +50,17 @@ public function updateProfileAction()
 
     /**
      * @AclAncestor("pim_user_user_edit")
+     *
+     * @param int $id
+     *
+     * @return JsonResponse|\Symfony\Component\HttpFoundation\Response
```
Comment (phaseinducer) [inline src/Oro/Bundle/UserBundle/Controller/UserController.php]: Import class please

File: src/Oro/Bundle/UserBundle/Controller/UserController.php [hunk 2]
```diff
@@ -50,9 +50,17 @@ public function updateProfileAction()
...
      */
-    public function apigenAction(User $user)
+    public function apigenAction($id)
     {
+        $userRepository = $this->container->get('pim_user.repository.user');
```
Comment (phaseinducer) [inline src/Oro/Bundle/UserBundle/Controller/UserController.php:61]: Is there a reason not to inject it ? (as there already is a dependency to our UserBundle)

Comment (jmleroux) [inline src/Oro/Bundle/UserBundle/Controller/UserController.php:61]: It is a Oro controller, extending `Symfony\Bundle\FrameworkBundle\Controller\Controller`

Comment (nidup) [inline src/Oro/Bundle/UserBundle/Controller/UserController.php:61]: ok, as discussed, no pb with this BC Break

**Comment by nidup** [comment]
GTM

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**